.github/workflows/pr-differences-mutants.yml

```
@@ -9,16 +9,43 @@ on:
       - ready_for_review
     paths:
       - '**.rs'
+  workflow_dispatch:
+    inputs:
+      ignore_timeout:
+        description: "Ignore mutants timeout limit"
+        required: false
+        type: choice
+        options:
+          - true
+        default: 'true'
 
 concurrency:
   group: pr-differences-${{ github.head_ref || github.ref || github.run_id }}
   # Always cancel duplicate jobs
   cancel-in-progress: true
 
 jobs:
+  check-right-permissions:
+    name: Check Right Permissions
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check Right Permissions To Trigger This
+        id: check_right_permissions
+        uses: stacks-network/actions/team-membership@main
+        with:
+          username: ${{ github.actor }}
+          team: 'blockchain-team'
+          GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
+
+      - name: Fail if the user does not have the right permissions
+        if: ${{ inputs.ignore_timeout == 'true' && steps.check_right_permissions.outputs.is_team_member != 'true' }}
```

If you make the change above to make this a boolean, then I think you need this corresponding change.
```suggestion
        if: ${{ inputs.ignore_timeout && steps.check_right_permissions.outputs.is_team_member != 'true' }}
```

---

i see. in that case then - why not set an env var after you check if the user is part of a team, and use that env var (our output var, either would work) in later conditionals?

that way, you could remove the workflow_dispatch choice option entirely.

---

that's a nice idea, but aren't the automatic workflow runs limited to 6 hours? the main goal would be to run for more than 6 hours and if that requires a manual trigger i think workflow_dispatch would be the easiest way to do this.

----
if the automatic workflow runs aren't limited to 6 hours, I would do these 2 actions:
1. give unlimited time for prs done by blockchain-team developers
2. keep the dispatch in case the pr is done by someone outside the blockchain-team so that someone from the team could trigger it manually, if that is the case

---

or are you referring specifically just to remove the true/false input on dispatch and let it take the context from who calls the workflow_dispatch? (true - part of blockchain-team, false - otherwise )

---

yes - if the person running this via workflow_dispatch is on the allowed team - we can bypass the time limit of 6 hours that we've set. 
else, we keep that time limit, and we can remove the choice option from displayed (we won't need it - either the calling actor is in the team or they aren't, and the time limit can set accordingly). 

so:
- workflow_dispatch called manually by someone *not on* allowed team: timeout at 6 hours
- workflow_dispatch called manually by someone *on* allowed team: no timeout set
- workflow triggered automatically (i.e. as part of a PR): timeout at 6 hours

---

i removed it, but there is an unknown. how does it has the context to differentiate between manual workflow_dispatch and automatic pr? (`2` and `3`)

previously i gave it this context through the input field that i added. on PRs it would be false and on workflow_dispatch true. now it proceeds on both cases the same way, running it as it sees that the action is done by a member of the allowed team, not knowing it should stop directly on the automatic PR trigger.

- this workflow dispatch work as expected https://github.com/stacks-network/stacks-core/actions/runs/10488274882
- this automatic PR trigger should have stopped directly and display that a manual trigger is required as it will timeout after 6 hours https://github.com/stacks-network/stacks-core/actions/runs/10488267327

---

> i removed it, but there is an unknown. how does it has the context to differentiate between manual workflow_dispatch and automatic pr? (`2` and `3`)
> 
> previously i gave it this context through the input field that i added. on PRs it would be false and on workflow_dispatch true. now it proceeds on both cases the same way, running it as it sees that the action is done by a member of the allowed team, not knowing it should stop directly on the automatic PR trigger.
> 
> * this workflow dispatch work as expected https://github.com/stacks-network/stacks-core/actions/runs/10488274882
> * this automatic PR trigger should have stopped directly and display that a manual trigger is required as it will timeout after 6 hours https://github.com/stacks-network/stacks-core/actions/runs/10488267327

to diff, you would need to modify the steps that run conditionally (or when you set an env var etc) by checking if the workflow was triggered manually (workflow_dispatch). ex:
```
github.event_name == 'workflow_dispatch'
```
https://github.com/stacks-network/stacks-core/blob/ci/custom-action-membership/.github/workflows/ci.yml#L116

once you know what triggered the workflow, you can adjust the environment as you need it. 

I *think* that's what you're asking at least. if not, ping me

---

awesome, i've missed that. i will update it

---

Worked as expected:
- stops on PR trigger on member if more mutants than threshold  https://github.com/stacks-network/stacks-core/actions/runs/10491998635
- running on workflow_dispatch CI https://github.com/stacks-network/stacks-core/actions/runs/10492055238